apps/recorder/widget.js

```diff
@@ -256,10 +256,36 @@
     options = options||{};
     if (isOn && !settings.recording) {
       var date=(new Date()).toISOString().substr(0,10).replace(/-/g,""), trackNo=10;
-      function getTrackFilename() { return "recorder.log" + date + trackNo.toString(36) + ".csv"; }
-      if (!settings.file || !settings.file.startsWith("recorder.log" + date)) {
-        // if no filename set or date different, set up a new filename
-        settings.file = getTrackFilename();
+      function nextTrackFilename() { return "recorder.log" + date + trackNo.toString(36) + ".csv"; }
+      function mostRecentFile() {
+        const logs = require("Storage")
+          .list(/^recorder\.log/)
+          .map(x => x.replace(/^recorder\.log/, ""));
+
+        if(logs.length > 0){
+          logs.sort();
+          const latest = "recorder.log" + logs[logs.length-1].replace("\1", "");
```

Comment:
Not sure why: above you remove recorder.log and here you add it again

Reply:
Ah duh - yes, I was thinking it'd impact the sort above, but it doesn't matter, shall fix